assets/app/views/_templateopt.html

```diff
@@ -22,11 +22,19 @@
           type="text"
           placeholder="{{ parameter | parameterPlaceholder }}"
           ng-model="parameter.value"
-          ng-required="parameter.required && !parameter.generate">
+          ng-required="parameter.required && !parameter.generate"
+          ng-trim="false">
     </span>
     <div class="help-block" ng-if="parameter.description">{{parameter.description}}</div>
     <div ng-show="paramForm[paramID].$error.required && paramForm[paramID].$touched" class="has-error">
-      <div class="help-block">{{parameter.name}} is required.</div>
+      <div class="help-block">{{parameter.displayName || parameter.name}} is required.</div>
+    </div>
+    <div ng-show="isOnlyWhitespace(parameter.value)" class="has-warning">
+      <span class="help-block">
+        The current value only has spaces.
```

Comment:
Without other changes, it's a bit awkward since the warning becomes an error saying the field is required. I'd need to look at that. Also we should probably focus the field after clearing its value.

![cursor_and_openshift_web_console](https://cloud.githubusercontent.com/assets/1167259/10171001/ba78f322-66a5-11e5-8ef4-9b0b341bb647.png)